### PR TITLE
Add DTN enum to Interop ComCtl32

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Add DTN enum to Interop ComCtl32.
- Remove DTN constants and replace their usages with the above enum values.
- Remove usused LPSTR_TEXTCALLBACK, STATUS_PENDING, FRERR_BUFFERLENGTHZERO and HDS_FULLDRAG constants.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2930)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DateTimePicker.cs [hunk 1]
```diff
@@ -1629,13 +1629,13 @@ private unsafe void WmReflectCommand(ref Message m)
                 User32.NMHDR* nmhdr = (User32.NMHDR*)m.LParam;
                 switch (nmhdr->code)
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/DateTimePicker.cs]: could cast this to `DTN`

File: src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            USERSTRING = USERSTRINGW,
+            WMKEYDOWNW = FIRST - 4,
+            WMKEYDOWN = WMKEYDOWNW,
+            FORMATW = FIRST - 3,
+            FORMAT = FORMATW,
+            FORMATQUERYW = FIRST - 2,
+            FORMATQUERY = FORMATQUERYW,
+            DROPDOWN = FIRST2 - 1,
+            CLOSEUP = FIRST2
+        }
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs]:
```suggestion
        public enum DTN
        {
            FIRST = 0 - 740,
            LAST = 0 - 745,
            FIRST2 = 0 - 753,
            LAST2 = 0 - 799,
            FORMATQUERYW = FIRST - 2,
            FORMATW = FIRST - 3,
            WMKEYDOWNW = FIRST - 4,
            USERSTRINGW = FIRST - 5,
            DATETIMECHANGE = FIRST2 - 6,
            CLOSEUP = FIRST2
            DROPDOWN = FIRST2 - 1,
        }
```

File: src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            FIRST = 0 - 740,
+            LAST = 0 - 745,
+            FIRST2 = 0 - 753,
+            LAST2 = 0 - 799,
+            DATETIMECHANGE = FIRST2 - 6,
+            USERSTRINGW = FIRST - 5,
+            USERSTRING = USERSTRINGW,
+            WMKEYDOWNW = FIRST - 4,
+            WMKEYDOWN = WMKEYDOWNW,
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs]: In some other enums we kept both. I have updated this and the other enum to remove them.

File: src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            LAST = 0 - 745,
+            FIRST2 = 0 - 753,
+            LAST2 = 0 - 799,
+            FORMATQUERYW = FIRST - 2,
+            FORMATW = FIRST - 3,
+            WMKEYDOWNW = FIRST - 4,
+            USERSTRINGW = FIRST - 5,
+            DATETIMECHANGE = FIRST2 - 6,
+            DROPDOWN = FIRST2 - 1,
+            CLOSEUP = FIRST2
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.DTN.cs]: And these? 😞 
```suggestion
            CLOSEUP = FIRST2
            DROPDOWN = FIRST2 - 1,
```

**Pull request merged**